src/base_directories.rs

```diff
@@ -952,6 +992,253 @@ mod test {
         });
     }
 
+    fn spawn_test_public_new() -> std::thread::JoinHandle<()> {
+        std::thread::spawn(|| {
+            let cwd = env::current_dir().unwrap().to_string_lossy().into_owned();
+            std::env::set_var("HOME", format!("{}/test_files/defaults", cwd));
+            let xd = BaseDirectories::new();
```

Comment:
This does a couple things:
- Tests `xdg::new()`
- Tests `HOME`
- Tests the behavior for missing `XDG_*` env vars

All of these were previously missing from the old tests.